staging/src/k8s.io/apiserver/pkg/storage/etcd3/store_test.go

```diff
@@ -106,15 +107,32 @@ func TestCreate(t *testing.T) {
 	if out.ResourceVersion == "" {
 		t.Errorf("output should have non-empty resource version")
 	}
+	if out.SelfLink != "" {
+		t.Errorf("output should have empty self link")
+	}
+
+	checkStorageInvariants(ctx, t, etcdClient, store, key)
+}
 
-	// verify that kv pair is not empty after set
-	getResp, err = etcdClient.KV.Get(ctx, key)
+func checkStorageInvariants(ctx context.Context, t *testing.T, etcdClient *clientv3.Client, store *store, key string) {
+	getResp, err := etcdClient.KV.Get(ctx, key)
 	if err != nil {
 		t.Fatalf("etcdClient.KV.Get failed: %v", err)
 	}
 	if len(getResp.Kvs) == 0 {
 		t.Fatalf("expecting non empty result on key: %s", key)
 	}
+	decoded, err := runtime.Decode(store.codec, getResp.Kvs[0].Value[5:])
```

Comment:
Why this is "Value[5:]"? What is stored in the first 5 chars? Can we rely on the fact that this is 5?